### PR TITLE
fix(language-core): infer define model type from options type

### PR DESCRIPTION
**Pull request by davidmatter**
If `T` in `defineModel<T>` is not defined, try to infer from user's code.

### CODE REVIEW

File: packages/language-core/lib/codegen/script/scriptSetup.ts [hunk 1]
```diff
@@ -323,6 +325,9 @@ function* generateComponentProps(
 			if (defineProp.type) {
 				yield scriptSetup.content.substring(defineProp.type.start, defineProp.type.end);
 			}
+			else if (defineProp.name && defineProp.nameIsString) {
+				yield `NonNullable<typeof ${propName}['value']>`;
+			}
 			else if (!defineProp.nameIsString) {
```
Comment (so1ve) [inline packages/language-core/lib/codegen/script/scriptSetup.ts:331]: Consider merging with this code branch?

Comment (davidmatter) [reply packages/language-core/lib/codegen/script/scriptSetup.ts:331]: What does this usecase actually cover? `!defineProp.nameIsString`

Comment (davidmatter) [reply packages/language-core/lib/codegen/script/scriptSetup.ts]: Merged branches

Comment (so1ve) [reply packages/language-core/lib/codegen/script/scriptSetup.ts]: Not sure tho 😂

**Comment by johnsoncodehk** [comment]
LGTM, thanks for your investigation!

**Pull request merged**